apps/web/src/pages/Pools/ViewPoolPage/components/ChangeDateDialog.tsx

```diff
@@ -0,0 +1,240 @@
+                idPrefix="type"
+                name="type"
+                data-h2-margin-bottom="base(x1)"
+                rules={{
+                  required: intl.formatMessage(errorMessages.required),
+                }}
+                legend={intl.formatMessage({
+                  defaultMessage: "Type of date change",
+                  id: "G6V8wJ",
+                  description:
+                    "Legend for the radio group to select whether to extend a date or set to now",
+                })}
+                items={[
+                  {
+                    label: intl.formatMessage({
+                      defaultMessage: "Extend closing date",
+                      id: "NTwwfL",
+                      description: "Option to extend the pool closing date",
+                    }),
+                    value: "extend",
+                  },
+                  {
+                    label: intl.formatMessage({
+                      defaultMessage: "Close immediately",
+                      id: "pXiwEH",
+                      description: "Option to close the pool",
+                    }),
+                    value: "close",
+                  },
+                ]}
+              />
+              {type === "extend" ? (
+                <DateInput
+                  id="expiryEndDate"
+                  legend={intl.formatMessage({
+                    defaultMessage: "End Date",
+                    id: "80DOGy",
+                    description:
+                      "Label displayed on the pool candidate form end date field.",
+                  })}
+                  name="expiryEndDate"
+                  rules={{
+                    required: intl.formatMessage(errorMessages.required),
+                    min: {
+                      value: strToFormDate(todayDate.toISOString()),
+                      message: intl.formatMessage(errorMessages.futureDate),
+                    },
+                  }}
+                />
+              ) : (
+                <>
+                  <p data-h2-margin-bottom="base(x1)">
+                    {intl.formatMessage({
+                      defaultMessage:
+                        "Please write the reason why this process is closing early.",
+                      id: "nqNCB/",
+                      description:
+                        "Helper message for closing the process early",
+                    })}
+                  </p>
+                  <TextArea
+                    id="reason"
+                    data-h2-margin-bottom="base(x.5)"
+                    label={intl.formatMessage({
+                      defaultMessage: "Reason for closing",
+                      id: "SQRzSl",
+                      description: "Label for the reason for closing the pool",
+                    })}
+                    name="reason"
+                    rules={{
+                      required: intl.formatMessage(errorMessages.required),
+                    }}
```

Comment:
String column maxes out at 256 characters I believe, so ought to squeeze in `wordLimit={256}`
Avoid rejected operations that way

Reply:
oh jeeze, so I agree with a word limit but I just realized that we should be using `text` instead of `string` here :sweat_smile: Thanks for spotting this!

Reply:
That works too 😛

Reply:
okay, updated with the new column type and word limit in [update reason to text column, set word limit](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9945/commits/a4fdb3daf3c5df55b0b11fabf7f35204d7a15387)